manifest/morph/morph.go

```diff
@@ -655,7 +660,10 @@ func morphObjectToType(v tftypes.Value, t tftypes.Type, p *tftypes.AttributePath
 			}
 			mvals[k] = nv
 		}
-		return tftypes.NewValue(t, mvals), diags
+		if diags := validateValue(t, mvals, p); diags != nil {
```

Comment:
I'm not sure what the purpose of this function is if newValue is going to call validateValue anyway?

Reply:
this might've been left when it came to figuring out why one of the manifest tests wasn't passing. Changed it to just using newValue() with checks all passing